File: gazda/nekretnine/report/transakcije_koje_pratimo/transakcije_koje_pratimo.py
# Copyright (c) 2025, Filip Ilic and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns = [
		{
			"fieldname": "nekretnina",
			"label": "Nekretnina",
			"fieldtype": "Link",
			"options": "Nekretnina",
			"width": 300
		},
		{
			"fieldname": "zakupac",
			"label": "Zakupac",
			"fieldtype": "Link",
			"options": "Osoba",
			"width": 200
		},
		{
			"fieldname": "transakcije",
			"label": "Transakcije",
			"fieldtype": "HTML",
			"width": 400
		}
	]
	
	data = []
	nekretnine = frappe.get_all(
		'Nekretnina',
		fields=['name', 'zakupac'],
		filters={'status': 'Zauzeto'}
	)
	
	for n in nekretnine:
		doc = frappe.get_doc('Nekretnina', n.name)
		transakcije = [t.transakcije for t in doc.pratimo_transakcije]
		pills = ' '.join([f'<span class="pill">{t}</span>' for t in transakcije]) if transakcije else ''
		data.append({
			'nekretnina': n.name,
			'zakupac': n.zakupac,
			'transakcije': f'<div class="pills-container">{pills}</div>'
		})
	
	return columns, data

File: gazda/nekretnine/report/transakcije_koje_pratimo/transakcije_koje_pratimo.js
// Copyright (c) 2025, Filip Ilic and contributors
// For license information, please see license.txt

frappe.query_reports["Transakcije koje pratimo"] = {
	"filters": [
		{
			"fieldname": "nekretnina",
			"label": __("Nekretnina"),
			"fieldtype": "Link",
			"options": "Nekretnina",
			"width": "300px"
		},
		{
			"fieldname": "zakupac",
			"label": __("Zakupac"),
			"fieldtype": "Link",
			"options": "Osoba",
			"width": "200px"
		},
		{
			"fieldname": "transakcije",
			"label": __("Pratimo ove transakcije"),
			"fieldtype": "Data",
			"width": "350px"
		}
	],
	onload: function(report) {
		// Add CSS for pills
		$('head').append(`
			<style>
				.pills-container {
					display: flex;
					flex-wrap: wrap;
					gap: 5px;
				}
				.pill {
					background-color: var(--orange-100);
					color: var(--orange-600);
					padding: 2px 8px;
					border-radius: 12px;
					font-size: 14px;
					white-space: nowrap;
				}
			</style>
		`);
	}
};
